src/Refinery/Tactic.hs

```diff
@@ -92,27 +150,27 @@ progress eq err t = do
   j <- goal
   a <- t
   j' <- goal
-  if j `eq` j' then pure a else throwError err
+  if j `eq` j' then pure a else failure err
 
 -- | @gather t f@ runs the tactic @t@, then runs @f@ with all of the generated subgoals to determine
 -- the next tactic to run.
-gather :: (MonadExtract ext m) => TacticT jdg ext err s m a -> ([(a, jdg)] -> TacticT jdg ext err s m a) -> TacticT jdg ext err s m a
+gather :: (MonadExtract ext err m) => TacticT jdg ext err s m a -> ([(a, jdg)] -> TacticT jdg ext err s m a) -> TacticT jdg ext err s m a
 gather t f = tactic $ \j -> do
     s <- get
     results <- lift $ proofs s $ proofState t j
-    msum $ flip fmap results $ \case
-        Left err -> throwError err
-        Right (_, _, jdgs) -> proofState (f jdgs) j
+    case results of
```

Comment:
What does this function do for partial proofs? Might be worth mentioning in the haddock

Reply:
Going to see if I can do this "right"